Chessington.GameEngine/Pieces/King.cs

```diff
@@ -10,7 +10,27 @@ public King(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            List<Square> availableMoves = new List<Square>();
+            
+            int[] row = new int[] { 1, 1, 1, 0, 0, -1, -1, -1};
+            int[] col = new int[] { -1, 0, 1, -1, 1, -1, 0, 1};
+            
+            for (int i = 0; i < 8; i++)
+            {
+                if(board.FindPiece(this).Row + row[i] >= 0
+                   && board.FindPiece(this).Row + row[i] < 8 
+                   && board.FindPiece(this).Col + col[i] >= 0 
+                   && board.FindPiece(this).Col + col[i] < 8 
+                   && (board.GetPiece(new Square(board.FindPiece(this).Row + row[i],
```

Comment:
I would also make helper board.IsSquareEmpty(Square) and board.IsSquareOccupiedByEnemy(Square), with names TBD, because it seems you reuse this logic in several places

Reply:
Or other helper function which is combination of the 2, if you think that's better in your case